core/src/test/java/fr/sncf/osrd/envelope_sim_infra/MRSPTest.java

```diff
@@ -125,14 +127,82 @@ public void mrspFromInfraPath() {
                 new TrackRangeView(15, 80, infra.getEdge("track", Direction.FORWARD))
         );
         var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
-
-        var mrsp = MRSP.from(path, testRollingStock, true, null);
+        var driverBehaviour = new DriverBehaviour(0, 0);
+        var mrsp = MRSP.from(path, testRollingStock, true, null, driverBehaviour);
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(15, 1));
         Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(29, 1));
         Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(30, 1));
         Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(49, 1));
         Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(51, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(56, 1));
+        Assertions.assertEquals(testRollingStock.maxSpeed, mrsp.interpolateSpeedRightDir(75, 1));
+    }
+
+    @Test
+    public void mrspWithDriverBehaviour() {
+        List<TrackRangeView> path = pathSenarioForTest();
+        var testRollingStock = TestTrains.VERY_SHORT_FAST_TRAIN;
+        var driverBehaviour = new DriverBehaviour(16, 17);
+        var mrsp = MRSP.from(path, testRollingStock, true, null, driverBehaviour);
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(0, 1));
+        Assertions.assertEquals(42, mrsp.interpolateSpeedRightDir(12, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(13, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(29, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(30, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(33, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(35, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(47, 1));
+        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(56, 1));
```

Comment:
There are many assert for the same speed limit. I don't think it's really necessary. What matters here is to test the width of the 21m/s speed limit, so making sure it spans between `30-17` and `50+16`
```suggestion
        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(13, 1));
        Assertions.assertEquals(21, mrsp.interpolateSpeedRightDir(66, 1));
```

Reply:
you're right